uk.co.compucorp.civicrm.hrleaveandabsences/js/src/leave-absences/leave-type-wizard/components/leave-type-wizard.component.js

```diff
@@ -96,26 +103,27 @@ define([
      * Checks if title provided is unique.
      * If existing titles are not yet loaded, the validation will not be executed,
      * instead, it will show that the field is "loading".
+     *
+     * @return {Promise}
      */
     function checkIfTitleIsUnique () {
       var titleField = vm.fieldsIndexed.title;
-      var title = titleField.value;
 
-      if (_.isEmpty(title)) {
-        return;
+      if (_.isEmpty(titleField.value) || titleField.waitsForSupportiveData) {
```

Comment:
`waitsForSupportiveData` is not representative of what this flag is about. You turn it `true` right before checking if its unique, you turn it `false` right after the check. Its value then has nothing to do on whether the "supportive data" is already there or not

As I suggested originally, please rename the property to something more descriptive like `validating`

Reply:
Done.